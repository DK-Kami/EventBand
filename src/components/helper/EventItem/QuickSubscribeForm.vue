<template>	
	<v-layout justify-center align-center>
    <v-dialog v-model="dialog" persistent max-width="50vw">
      <template #activator="{ on }">
        <v-btn block  color="primary" dark v-on="on">
					{{ isLogged ? 'Book a ticket' : 'Quck subscribe form'}}
				</v-btn>
      </template>
      <v-card>
				<v-layout v-if="!isLogged" justify-center>
					<v-card-title class="title">For quick subscribe fill the following fields</v-card-title>
				</v-layout>
				<v-layout
					justify-center
				>
					<form-base 
						ref="quickForm"
						@submit.prevent="subscribeOnEvent">
						<template #title>
							Quick subscribe form
						</template>
						<v-layout
							v-if="!isLogged"
							align-content-center
							justify-center
							column
						> 
							<v-text-field
								v-model="email"
								:rules="[rules.email, rules.required]"
								label="Your e-mail"
								class="width-l"
								outlined
								dense
							/>
							<v-text-field
								v-model="surname"
								:rules="[rules.required, rules.min('surname')]"
								label="Your surname"
								class="width-l"
								maxlength="20"
								type="email"
								outlined
								dense
							/>
							<v-text-field
								v-model="name"
								:rules="[rules.required, rules.min('name')]"
								label="Your name"
								class="width-l"
								maxlength="20"
								outlined
								dense
							/>
						</v-layout>
						<v-layout column>
							<template v-if="!tickets.length">
								<span class="title align-self-center"> No tickets this event</span>
							</template>
							<template v-else>
								<span class="headline align-self-center" > Tickets </span>
								<v-radio-group v-model="radio">
									<v-tooltip 
										v-for="ticket in tickets"
										:key="ticket.uuid"
										left
									>
										<template #activator="{ on }">
											<v-radio 
												color="black"
												:label="ticket.name"
												:value="ticket.uuid"
												v-on="on"
											/>
										</template>
										<span> {{ ticket.description }} </span>
									</v-tooltip>
								</v-radio-group>
							</template>
						</v-layout>
						<template #actions>
							<v-layout 
								fill-width
								justify-center
							>
								<v-btn color="error" text @click="dialog = false">Back</v-btn>
								<v-btn type="submit" color="green darken-1 white--text">Subscribe</v-btn>
							</v-layout>
						</template>
					</form-base>
				</v-layout>
      </v-card>
    </v-dialog>
		<v-dialog 
			v-model="cbDialog"
			width="400px"
			class="pa-0"
		>
			<v-alert
				border="top"
				type="info"
				class="ma-0"
			> 
				{{ cbMessage }} 
			</v-alert>
		</v-dialog>
	</v-layout>
</template>

<script>
import FormBase from '@/components/base/FormBase';

export default {
	name: 'QuickSubscribeForm',

	components: {
		FormBase,
	},
	
	props: {
		tickets: Array,
	},

	data: (vm) => ({
		dialog: false,
		cbDialog: false,
		email: '',
		surname: '',
		name: '',
		radio: vm.tickets[0].uuid,

		cbMessage: 'nothing happend',
	}),

	computed: {
		rules() {
			return this.$store.getters['getRules'];
		},
		isLogged() {
			return this.$store.getters['auth/getIsLogged'];
		},
	},

	methods: {
		async subscribeOnEvent() {
			if (!this.radio) {
				this.$refs.quickForm.setAlert('Choose ticket', 'warning');
				return;
			};

			if (this.isLogged) {
				let message = await this.$store.dispatch('event/subscribeOnEvent', this.radio);
				this.$store.dispatch('notification/set', message);
				return;
			}

			let message = await this.$store.dispatch('nonauth/subscribeOnEvent', {
				ticketUuid: this.radio,
				email: this.email,
				surname: this.surname, 
				name: this.name,
			});
			
			this.$refs.quickForm.resetForm();
			this.radio = this.tickets[0].uuid;
			this.$store.dispatch('notification/set', {
				message: message,
				type: 'info',
			});
			this.dialog = false;
		},

	},
}
</script>

<style src="@/scss/_width.scss" lang="scss">
	
</style>