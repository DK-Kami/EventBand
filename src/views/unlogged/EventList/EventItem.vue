<template>
	<v-card
		class="pa-2 my-2 d-inline-block"
		style="max-width: 65vw; min-width: 65vw"
		color="secondary"
		elevation="0"
		tile
	>
		<v-container>
			<v-layout row class="ma-0 px-3" align-center>
				<v-flex
					xl1 lg1
				>
					<v-img
						height="100"
						width="100"
						:src="imagePath"/>
				</v-flex>
				<v-flex xl4 lg4 md4>
					<v-layout
						class="d-flex flex-column px-3">
						<p class="text-left white--text headline" > {{ eventName }} </p>
						<p class="text-left white--text" > {{ eventDescription }} </p>
					</v-layout>
				</v-flex>
				<v-flex 
					xl2 lg2 
					style="max-height: 130px; overflow-y: auto;"
				>
					<v-layout
						class="d-flex flex-column"
						style="">
						<v-chip
							v-for="tag in tags"
							:key="tag.text"
							color="primary"
							class="my-1"
							small
						>
							<v-icon
								left
								v-text="tag.icon"
							></v-icon>
							{{ tag.text }}
						</v-chip>
					</v-layout>
				</v-flex>
				<v-flex xl2 lg2 class="px-2 pt-1 white--text">
					<p> Price:  {{ price }} $ </p>
					<p> Participants: {{ currentParticipantsAmount }} / {{ maxParticipantsAmount }} </p>
					<p> Date: <v-btn class="white--text" text v-text="date" /> Time: <v-btn class="white--text" text v-text="time" /> </p>
				</v-flex>
				<v-flex xl3 lg3>
					<v-card-actions>
						<v-layout  class="d-flex flex-column justify-start py-3">
							<v-btn 
								color="primary"
								class="mb-1"
								depressed
								x-large
							> subscribe 
							</v-btn>
							<v-btn 
								color="secondary"
								class="my-1"
								depressed
							> view 
							</v-btn>
							<v-btn 
								color="primary"
								class="my-1"
								text
								> don`t show again 
							</v-btn>
						</v-layout>
					</v-card-actions>
				</v-flex>
			</v-layout>
		</v-container>
	</v-card>
</template>

<script>
import CalendarInput from '@/components/helper/DateTimePicker/CalendarInput';
import TimeInput from '@/components/helper/DateTimePicker/TimeInput';

const items = [
	{
		text: '3D-modeling',
		icon: 'mdi-desktop-mac-dashboard',
	},
	{
		text: 'Science',
		icon: 'mdi-atom',
	},
	{
		text: 'Music',
		icon: 'mdi-guitar-acoustic',
	},
	{
		text: 'Sport',
		icon: 'mdi-volleyball',
	},
	{
		text: 'Art',
		icon: 'mdi-brush',
	},
	{
		text: 'Religion',
		icon: 'mdi-christianity',
	},
];

export default {
	name: 'EventItem',

	components: {
		CalendarInput,
		TimeInput,
	},
	
	data: () => ({
		path: '/logo-organization.png',
		name: 'Event name',
		price: '20',
		date: '4.21.2020',
		time: '01:25 PM',
		currentParticipantsAmount: 35,
		maxParticipantsAmount: 100,
		description: `Lorem ipsum dolor sit amet consectetur, adipisicing elit. Rem illo ducimus quas praesentium repellendus. Itaque, officiis dolor corrupti`,
		items,
	}),

	computed: {
		imagePath() {
			return this.path; 
		},
		eventName() {
			return this.name;
		},
		eventDescription() {
			return this.description;
		},
		tags() {
			return this.items;
		},
		eventDate() {
			return this.date;
		},
		eventTime() {
			return this.time;
		}
	},
}
</script>

<style lang="scss" scoped>
	::-webkit-scrollbar { 
    width: 0px;  
    background: transparent;
	}
</style>