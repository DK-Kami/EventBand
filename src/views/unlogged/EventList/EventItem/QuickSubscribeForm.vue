<template>	
	<v-row justify="center" >
    <v-dialog v-model="dialog" persistent max-width="50vw">
      <template #activator="{ on }">
        <v-btn color="primary" dark v-on="on">Quick subscribe</v-btn>
      </template>
      <v-card>

						<v-layout justify-center>
							<v-card-title class="title">For quick subscribe fill the following fields</v-card-title>
						</v-layout>
						<v-layout
							justify-center
						>
								<form-base 

									@submit.prevent="subscribeOnEvent">
									<template #title>
										Quick subscribe form
									</template>
									<v-layout
										column
										align-content-center
										justify-center
									> 
										<v-text-field
											v-model="email"
											:rules="[rules.email, rules.required]"
											label="Your e-mail"
											class="width-l"
											outlined
											dense
										/>
										<v-text-field
											v-model="surname"
											:rules="[rules.required, rules.min('surname')]"
											label="Your surname"
											class="width-l"
											maxlength="20"
											type="email"
											outlined
											dense
										/>
										<v-text-field
											v-model="name"
											:rules="[rules.required, rules.min('name')]"
											label="Your name"
											class="width-l"
											maxlength="20"
											outlined
											dense
										/>
									</v-layout>
									<template #actions>
										<v-layout 
											fill-width
											justify-center
										>
											<v-btn color="error" text @click="dialog = false">Back</v-btn>
											<v-btn type="submit" color="green darken-1 white--text">Subscribe</v-btn>
										</v-layout>
									</template>
								</form-base>
						</v-layout>
      </v-card>
    </v-dialog>
  </v-row>
</template>

<script>
import FormBase from '@/components/base/FormBase';

export default {
	name: 'QuickSubscribeForm',
	components: {
		FormBase,
	},
	data: () => ({
		dialog: false,
		email: '',
		surname: '',
		name: '',
	}),
	computed: {
		rules() {
			return this.$store.getters['getRules'];
		},
	},

	methods: {
		subscribeOnEvent() {
			console.log(lol);
			
		}
	},
}
</script>

<style src="@/scss/_width.scss" lang="scss">
	
</style>