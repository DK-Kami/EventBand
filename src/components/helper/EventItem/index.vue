<template>
	<v-card
		class="pa-2 my-2 custom-elevation"
		outlined
	>

		<v-layout>
			<!-- {{logo || event.organization.logo}} -->
			<v-flex xs1>
				<v-avatar size="128px">
					<v-img :src="`https://event-band-api.ru:5000/static/organizations/${logo || event.organization.logo}`" />
				</v-avatar>
			</v-flex>

			<v-flex xs3 class="pl-4">
				<v-layout column>
					<v-card-title class="headline" > {{ event.name }} </v-card-title>
					<v-card-subtitle class="subtitle-1 pa-0 pl-4 text-truncate">
						{{ event.description }}
					</v-card-subtitle>
				</v-layout>
			</v-flex>
			<v-flex xs2>
				<div class="text-center "> Tags </div>
				<v-layout wrap>
					<v-chip
						v-for="(tag, i) in tags"
						:key="i" 
						class="ma-1"
						small
					> 
						{{ tag.name || tag }} 
					</v-chip>
				</v-layout>
			</v-flex>
			<v-flex xs3 class="left-border pl-4">
				<v-layout column>
					<span class="align-self-center"> Info </span>
					<div> Max tickets amount 
						<span class="orange--text text--darken-4"> ({{ event.count }}) </span>
					</div>
					<div> Min ticket price 
						<span class="green--text text--darken-2"> ({{ minimumPrice }}) </span>  
					</div>
					<div> Starting
						<span class="pink--text text--darken-4"> {{ dateFrom }} </span>
					</div>
					<div> Ending
						<span class="pink--text text--darken-4"> {{ dateTo }} </span>
					</div>
				</v-layout>
			</v-flex>
			<v-flex xs3>
				<v-layout fill-height wrap align-center>
					<quick-subscribe-form 
						v-if="canSubscribe"
						:tickets="tickets"/>
					<v-btn
						v-if="canView"
						:to=" isOrg ? `/my-organization/event/${event.uuid}` : `/event/${event.uuid}`"
						color="accent"
						block
					>
						View
					</v-btn>
					<v-btn
						v-if="canEdit"
						color="secondary"
						block
						@click="openEditForm"
					>
						Edit event
					</v-btn>
					<v-btn
						v-if="canEdit"
						color="secondary darken-3"
						block
						@click="openChatEditPage"
					>
						Edit chat
					</v-btn>
				</v-layout>
			</v-flex>
		</v-layout>
	</v-card>
</template>

<script>
import CalendarInput from '@/components/helper/DateTimePicker/CalendarInput';
import TimeInput from '@/components/helper/DateTimePicker/TimeInput';
import QuickSubscribeForm from './QuickSubscribeForm';

export default {
	name: 'EventItem',

	components: {
		QuickSubscribeForm,
		CalendarInput,
		TimeInput,
	},

	props: {
		tickets: Array,
		event: Object,
		tags: Array,
		logo: {
			type: String,
			default: undefined,
		},

		canSubscribe: true,
		canView: true,
		canEdit: false,
	},

	created() {
		console.log(this.event);		
	},
	
	data: () => ({
		path: 'logo-organization.png',
	}),

	computed: {
		dateFrom() {
			return this.event.datetimeFrom.split('T')[0];
		},
		dateTo() {
			return this.event.datetimeTo.split('T')[0];
		},
		imagePath() {
			return this.path; 
		},
		isLogged() {
			return this.$store.getters['auth/getIsLogged'];
		},

		isOrg() {
			return this.$store.getters['auth/getIsLoggedAsOrganization'];
		},

		minimumPrice() {
			if (this.event.price === undefined) {
				return this.event.minPrice;
			} else {
				return this.event.price;
			}
		},
	},

	methods: {
		openEditForm() {
			this.$store.dispatch('organization/setEditableEvent', {
				uuid: this.event.uuid,
				name: this.event.name,
				description: this.event.description,
				datetimeFrom: this.event.datetimeFrom,
				datetimeTo: this.event.datetimeTo,
				coords: this.event.coords,
				count: this.event.count || 1,
				tags: this.tags,
				tickets: this.tickets,
			});
			this.$store.dispatch('organization/setIsOnEditTrue');
			this.$router.push('/my-organization/event/create');
		},

		openChatEditPage() {
			this.$store.dispatch('eventchat/setChatUuid', this.event.chatUuid);
			this.$router.push('/my-organization/chat/edit');
		},
	},
}
</script>

<style lang="scss" scoped>

	.left-border {
		border-left: 2px solid gray;
	}
	
</style>