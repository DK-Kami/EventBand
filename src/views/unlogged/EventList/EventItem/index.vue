<template>
	<v-card
		class="pa-2 my-2 d-inline-block"
		color="secondary"
		elevation="0"
		tile
	>
		<v-container>
			<v-layout row class="ma-0 px-3" align-center>
				<v-flex
					xl1 lg1
				>
					<v-img
						height="100"
						width="100"
						:src="imagePath"/>
				</v-flex>
				<v-flex xl4 lg4 md4>
					<v-layout
						class="d-flex flex-column px-3">
						<p class="text-left white--text headline" > {{ eventName }} </p>
						<p class="text-left white--text" > {{ eventDescription }} </p>
					</v-layout>
				</v-flex>
				<v-flex 
					xl2 lg2 
				>
					<v-layout wrap >
						<v-flex xs12 class="white--text mt-n4">
							Tags	
						</v-flex>
						<v-flex 
							style="overflow-y: auto; max-height: 130px;"
							shrink
						>
							<v-chip
								v-for="tag in tags"
								:key="tag.text"
								color="primary"
								class="my-1 mr-1"
								small
							>
								<v-icon
									v-text="tag.icon"
									small
									left
								></v-icon>
								{{ tag.text }}
							</v-chip>
						</v-flex>
					</v-layout>
				</v-flex>
				<v-flex xl2 lg2 class="pl-4 px-2 pt-1 white--text">
					<p> Price:  {{ price }} $ </p>
					<p> Participants: {{ currentParticipantsAmount }} / {{ maxParticipantsAmount }} </p>
					<div> Date: <v-btn class="white--text" text v-text="date" /> </div> <div> Time: <v-btn class="white--text" text v-text="time" /> </div> 
				</v-flex>
				<v-flex xl3 lg3>
					<v-card-actions>
						<v-layout  class="d-flex flex-column justify-start py-3">
							<template v-if="isLogged" >
								<v-btn 
									color="primary"
									class="mb-1"
									depressed
									x-large
									@click="subsribeOnEvent"
								> subscribe 
								</v-btn>
							</template>
							<template v-else>
								<quick-subscribe-form/>
							</template>
							<v-btn 
								color="secondary"
								class="my-1"
								depressed
							> view 
							</v-btn>
							<v-btn 
								color="primary"
								class="my-1"
								text
								> don`t show again 
							</v-btn>
						</v-layout>
					</v-card-actions>
				</v-flex>
			</v-layout>
		</v-container>
	</v-card>
</template>

<script>
import CalendarInput from '@/components/helper/DateTimePicker/CalendarInput';
import TimeInput from '@/components/helper/DateTimePicker/TimeInput';
import QuickSubscribeForm from './QuickSubscribeForm';

const items = [
	{
		text: '3D-modeling',
		icon: 'mdi-desktop-mac-dashboard',
	},
	{
		text: 'Science',
		icon: 'mdi-atom',
	},
	{
		text: 'Music',
		icon: 'mdi-guitar-acoustic',
	},
	{
		text: 'Sport',
		icon: 'mdi-volleyball',
	},
	{
		text: 'Art',
		icon: 'mdi-brush',
	},
	{
		text: 'Religion',
		icon: 'mdi-christianity',
	},
];

export default {
	name: 'EventItem',

	components: {
		CalendarInput,
		TimeInput,
		QuickSubscribeForm,
	},
	
	data: () => ({
		path: '/logo-organization.png',
		name: 'Event name',
		price: '20',
		date: '4.21.2020',
		time: '01:25 PM',
		currentParticipantsAmount: 35,
		maxParticipantsAmount: 100,
		description: `Lorem ipsum dolor sit amet consectetur, adipisicing elit. Rem illo ducimus quas praesentium repellendus. Itaque, officiis dolor corrupti`,
		items,
	}),

	computed: {
		imagePath() {
			return this.path; 
		},
		eventName() {
			return this.name;
		},
		eventDescription() {
			return this.description;
		},
		tags() {
			return this.items;
		},
		eventDate() {
			return this.date;
		},
		eventTime() {
			return this.time;
		},
		isLogged() {
			return this.$store.getters['getIsLogged'];
		},
	},
}
</script>

<style lang="scss" scoped>
	// ::-webkit-scrollbar { 
  //   width: 0px;  
  //   background: transparent;
	// }
</style>