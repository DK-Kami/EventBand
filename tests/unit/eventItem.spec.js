import {
	mount,
	shallowMount
} from './index';

import EventItem from '@/components/helper/EventItem';

describe('Event item - компонент для вывода мероприятий', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(EventItem, {
			store: false,
			propsData: {

				"tickets": [
					{
						"uuid": "03b303f8-cff9-4e3d-a90f-4ba926188f53",
						"name": "Welcome to MoscowJS 47",
						"description": "",
						"count": 250,
						"price": 0
					}
				],
				"event": {
					"uuid": "300921b1-4ae3-442d-8396-642cde5e565b",
					"name": "MoscowJS 47",
					"description": "MoscowJS return! MoscowJS 47",
					"coords": "(55.7522200, 37.5155600)",
					"datetimeFrom": "2020-05-10T00:00:00.000Z",
					"datetimeTo": "2020-05-10T00:00:00.000Z",
					"count": 250,
					"price": 0,
					"organization": {
							"uuid": "2dabf617-911f-4da8-85ee-d1a03685ba82",
							"name": "Tatar",
							"reputation": 7,
							"logo": "tatar.jpg"
					},
				},
				"tags": [
					{
							"id": 2,
							"name": "Technologies"
					},
					{
							"id": 3,
							"name": "IT"
					},
					{
							"id": 16,
							"name": "Alcoparty"
					},
				],
			},
		});
	});

	it ('Делаем снапшот', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Event item является компонентом Vue', () => {
		expect(wrapper.isVueInstance()).toBeTruthy();
	});

	it('Имеет avatar', () => {
		expect(wrapper.find('.v-avatar').exists()).toBeTruthy();
	});

	it('Card имеет заголовок', () => {
		expect(wrapper.find('.v-card__title').exists()).toBeTruthy();
	});

	it('Card имеет подзаголовок', () => {
		expect(wrapper.find('.v-card__subtitle').exists()).toBeTruthy();
	})

	it('Имеет контейнер для тегов', () => {
		expect(wrapper.find('div > Tags')).toBeTruthy();
	});

	it('Имеет контейнер для информации о мероприятии', () => {
		expect(wrapper.find('span > Info')).toBeTruthy();
	});
});