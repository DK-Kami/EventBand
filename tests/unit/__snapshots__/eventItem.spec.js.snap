// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event item - компонент для вывода мероприятий Делаем снапшот 1`] = `
<div class="pa-2 my-2 custom-elevation v-card v-card--outlined v-sheet theme--light">
  <div class="layout">
    <div class="flex xs1">
      <div class="v-avatar" style="height: 128px; min-width: 128px; width: 128px;">
        <div class="v-responsive v-image">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
          <div class="v-responsive__content"></div>
        </div>
      </div>
    </div>
    <div class="flex pl-4 xs3">
      <div class="layout column">
        <div class="v-card__title headline"> MoscowJS 47 </div>
        <div class="v-card__subtitle subtitle-1 pa-0 pl-4 text-truncate">
          MoscowJS return! MoscowJS 47
        </div>
      </div>
    </div>
    <div class="flex xs2">
      <div class="text-center "> Tags </div>
      <div class="layout wrap"><span class="ma-1 v-chip v-chip--no-color theme--light v-size--small"><span class="v-chip__content"> 
					Technologies 
				</span></span><span class="ma-1 v-chip v-chip--no-color theme--light v-size--small"><span class="v-chip__content"> 
					IT 
				</span></span><span class="ma-1 v-chip v-chip--no-color theme--light v-size--small"><span class="v-chip__content"> 
					Alcoparty 
				</span></span></div>
    </div>
    <div class="flex left-border pl-4 xs3">
      <div class="layout column"><span class="align-self-center"> Info </span>
        <div> Max tickets amount
          <span class="orange--text text--darken-4"> (250) </span></div>
        <div> Min ticket price
          <span class="green--text text--darken-2"> (0) </span></div>
        <div> Starting
          <span class="pink--text text--darken-4"> 2020-05-10 </span></div>
        <div> Ending
          <span class="pink--text text--darken-4"> 2020-05-10 </span></div>
      </div>
    </div>
    <div class="flex xs3">
      <div class="layout fill-height wrap align-center">
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
